currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -227,7 +228,30 @@ def _update(self, date_from, date_to, newest_only=False):
                         })
 
             if is_scheduled:
-                provider._schedule_next_run()
+                # Search if every records (all dates and all currencies) are save
+                dates_need_check = []
+                delta = date_to - date_from
+                for i in range(delta.days + 1):
+                    day = date_from + relativedelta(days=i)
+                    dates_need_check.append(day.strftime("%Y-%m-%d"))
+
+                total_records = CurrencyRate.search_count([
+                    ('company_id', '=', provider.company_id.id),
+                    ('currency_id', 'in', provider.currency_ids.ids),
+                    ('name', 'in', dates_need_check)
+                ])
+                total_correct = len(provider.currency_ids.ids) * len(dates_need_check)
+                if total_records == total_correct:
+                    provider._schedule_next_run()
+                else:
+                    _logger.info(
```

Comment:
Maybe this should be a warning?

```suggestion
                    _logger.warning(
```